core/src/main/java/kafka/server/SharePartition.java

```diff
@@ -428,6 +452,19 @@ public CompletableFuture<Optional<Throwable>> acknowledge(String memberId, List<
         Throwable throwable = null;
         lock.writeLock().lock();
         List<InFlightState> updatedStates = new ArrayList<>();
+        boolean canMoveStartOffset = false;
+        AcknowledgementBatch firstBatch = acknowledgementBatch.get(0);
```

Comment:
It would be better to use an iterator of some kind rather than using get(int) on acknowledgementBatch.